ingestion/src/metadata/ingestion/source/dashboard/mode.py

```diff
@@ -0,0 +1,231 @@
+        self.base_url = "https://app.mode.com"
+        self.data_sources = self.client.get_all_data_sources(self.workspace_name)
+
+    @classmethod
+    def create(cls, config_dict, metadata_config: OpenMetadataConnection):
+        """Instantiate object
+        Args:
+            config_dict:
+            metadata_config:
+        Returns:
+            ModeSource
+        """
+        config = WorkflowSource.parse_obj(config_dict)
+        connection: ModeConnection = config.serviceConnection.__root__.config
+        if not isinstance(connection, ModeConnection):
+            raise InvalidSourceException(
+                f"Expected ModeConnection, but got {connection}"
+            )
+        return cls(config, metadata_config)
+
+    def get_dashboards_list(self) -> Optional[List[dict]]:
+        """
+        Get List of all dashboards
+        """
+        self.dashboards = self.client.fetch_all_reports(self.workspace_name)
+        return self.dashboards
+
+    def get_dashboard_name(self, dashboard_details: dict) -> str:
+        """
+        Get Dashboard Name
+        """
+        return dashboard_details.get("name")
+
+    def get_dashboard_details(self, dashboard: dict) -> dict:
+        """
+        Get Dashboard Details
+        """
+        return dashboard
+
+    def get_dashboard_entity(self, dashboard_details: dict) -> CreateDashboardRequest:
+        """
+        Method to Get Dashboard Entity, Dashboard Charts & Lineage
+        """
+        yield CreateDashboardRequest(
+            name=dashboard_details.get("token"),
+            # Mode has no hostPort property. All URL details are present in the webUrl property.
+            dashboardUrl=dashboard_details["_links"]["web"]["href"],
+            displayName=dashboard_details.get("name"),
+            description=dashboard_details.get("description")
+            if dashboard_details.get("description")
+            else "",
+            charts=get_chart_entities_from_id(
+                chart_ids=self.charts,
+                metadata=self.metadata,
+                service_name=self.config.serviceName,
+            ),
+            service=EntityReference(id=self.service.id, type="dashboardService"),
+        )
+
+    def get_lineage(self, dashboard_details: Any) -> Optional[AddLineageRequest]:
+        """
+        Get lineage between dashboard and data sources
+        """
+        try:
+            response_queries = self.client.get_all_queries(
+                workspace_name=self.workspace_name,
+                report_token=dashboard_details["token"],
+            )
+            queries = response_queries["_embedded"]["queries"]
+            for query in queries:
+                data_source_id = query.get("data_source_id")
+                if not data_source_id:
+                    continue
+                data_source = self.data_sources.get(data_source_id)
+                table_list = Parser(query.get("raw_query"))
```

Comment:
Could we use the `sqllineage.runner.LineageParser` here? We are removing the sql-metadata one. You should be able to pick up the same information by retrieving `parser.source_tables`

Reply:
okay I'll use the new one